src/engine/Range_with_metavars.ml

```diff
@@ -65,6 +65,15 @@ let included_in config rv1 rv2 =
      |> List.for_all (fun (mvar, mval1) ->
             match List.assoc_opt mvar rv2.mvars with
             | None -> true
+            (* Capture group metavariables (of the form $1, $2, etc) may
+               be introduced implicitly via regular expressions that have
+               capture groups in them.
+               The unification of these metavariables can be dangerous, as
+               it will prevent matches, when users may not even know these
+               metavariables exist.
+               To be safe, let's assume they always unify.
```

Comment:
You can maybe add a note like
* note that this does not concern named-capture-group-metavars introduced via the <?xxx> regexp syntax. 
* Those are still unified (as they are explicitely named)